src/nvim/api/extmark.c

```diff
@@ -105,7 +105,24 @@ bool ns_initialized(uint32_t ns)
   return ns < (uint32_t)next_namespace_id;
 }
 
-static Array extmark_to_array(const ExtmarkInfo *extmark, bool id, bool add_dict)
+static void insert_highlight(void *dict, char *key, int hl_id, bool hl_name)
```

Comment:
This could be simplified to return an Object and just return either `STRING_OBJ(cstr_to_string(syn_id2name(hl_id)))` or `INTEGER_OBJ(hl_id)`. the PUT or ADD part should be done by the caller.